src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/TrustDetailsValidator.java

```diff
@@ -114,16 +138,33 @@ private void validateCeasedDate(OverseasEntitySubmissionDto overseasEntitySubmis
         }
     }
 
-    private boolean isTrustStillRequired(OverseasEntitySubmissionDto overseasEntitySubmissionDto, String trustId, boolean istrustInvolvedInOverseasEntity) {
+    private boolean noBeneficalOwenersArePresent(OverseasEntitySubmissionDto overseasEntitySubmissionDto) {
+        boolean allIndividualsDisassociated = overseasEntitySubmissionDto.getBeneficialOwnersIndividual() != null
+                && overseasEntitySubmissionDto.getBeneficialOwnersIndividual().stream().anyMatch(
+                boIndividualDto ->
+                        (boIndividualDto.getTrustIds() == null || boIndividualDto.getTrustIds().isEmpty())
+                        || boIndividualDto.getCeasedDate() != null
+                        || boIndividualDto.getTrusteesNatureOfControlTypes() == null
+                        || boIndividualDto.getTrusteesNatureOfControlTypes().isEmpty());
+
+
+        boolean allCoporatesDisassociated = overseasEntitySubmissionDto.getBeneficialOwnersCorporate() != null
+                && overseasEntitySubmissionDto.getBeneficialOwnersCorporate().stream().anyMatch(
+                boCorporateDto ->
+                        (boCorporateDto.getTrustIds() == null || boCorporateDto.getTrustIds().isEmpty())
+                        || boCorporateDto.getCeasedDate() != null
+                        || boCorporateDto.getTrusteesNatureOfControlTypes() == null
+                        || boCorporateDto.getTrusteesNatureOfControlTypes().isEmpty());
+
+        return allIndividualsDisassociated && allCoporatesDisassociated;
+    }
+
+    private boolean isTrustStillRequired(OverseasEntitySubmissionDto overseasEntitySubmissionDto, String trustId, Boolean istrustInvolvedInOverseasEntity) {
         // Use the trust id to whizz through all the Beneficial Owners that are associated with this trust. As soon as a
         // matching BO is found where cease date is null and a trust NOC is set then this indicates that the trust is
         // still required
 
-        if (!istrustInvolvedInOverseasEntity) {
-           return false;
-        }
-
-        if (trustId == null) {
+        if ((istrustInvolvedInOverseasEntity != null && !istrustInvolvedInOverseasEntity) || trustId == null) {
```

Comment:
I think you can write `Boolean.FALSE.equals(istrustInvolvedInOverseasEntity)` in place of `(istrustInvolvedInOverseasEntity != null && !istrustInvolvedInOverseasEntity)` here. Takes care of the `null` check. Hopefully your unit tests exercise both scenarios to give confidence in the refactoring.

Reply:
Updated this